enhancements/installer/kubevirt-ipi.md

```diff
@@ -0,0 +1,346 @@
+tenant cluster VMs. See the [baremetal ipi networking doc][baremetal-ipi-networking]
+
+ 
+
+### Implementation Details/Notes/Constraints [optional]
+
+1. Survey
+
+    The installation starts and right after the user supplies their public ssh key,
+and then chooses `KubeVirt` the installation will ask for all the relevant details
+of the installation: **kubeconfig** for the infrastructure OpenShift, **namespace**, **storageClass**, 
+ **networkName (NAD)** and other KubeVirt specific attributes. 
+The installer will validate it can communicate with the api, otherwise it will fail to proceed.
+
+   With that the survey continues to the general cluster name, domain name, and 
+the rest of the non-KubeVirt specific question.
+
+2. Resource creation - terraform
+
+   Terraform uses kubernetes provider to create:
+
+    - [DataVolume CR][data-volumes] with RHCOS image
+
+         *Note:* In disconnected environment the user will need to provide a local image that the installer
+          can upload to the namespace.
+    - secrets for the Ignition configs of the VMs
+    - 1 bootstrap machine
+    - 3 masters
+    
+    Only on network option 1 - pods network:
+    - Services and routes for DNS and LB
+
+3. Bootstrap
+
+    The bootstrap VM has a huge Ignition config set using terraform as secrets and is visible
+as secrets on the infra OpenShift. KubeVirt boots that VM with that content as ConfigDrive and the 
+bootstraping begins when the `bootkube.service` systemd service starts.
+
+    This process described more thoroughly in the [installer overview document][https://github.com/OpenShift/installer/blob/37b99d8c9a3878bac7e8a94b6b0113fad6ffb77a/docs/user/overview.md#cluster-installation-process]
+
+4. Masters bootstrap
+
+    Master VMs are booting using a stub Ignition config that are waiting early in
+the Ignition service to load their Ignition config from a URL. That URL is the
+ `https://<internal-api-vip>/config/master` which is still not available until
+the **bootstrap** VM is exposing it. It takes few minutes till it does.
+
+    When the MachineConfigServer is available on the bootstrap, the masters pull their Ignition config 
+    and boot up joining the tenant cluster as masters and start scheduling pods.
+
+5. Workers bootstrap
+
+    After the masters and the control plane is up, we will scale the MachineSet to create workers
+    by the machine-api-operator.
+
+
+### Risks and Mitigations
+
+- Network
+
+    - Pods network option
+        -(OCP gap) The ports 22623/22624 that are used by MCS are blocked on the 
+        pods network and prevent from the nodes to pull Ignition config and updates.
+        - (KubeVirt gap) Interface binding - Currently the only supported binding on the pods
+        network is masquerade which means that all nodes are behind NAT, each VM
+        behind the NAT of his own pod.
+        - (OpenShift/KubeVirt gap) Static IP - OpenShift assumes that node's IP addresses are static,
```

Comment:
related topic, some customers insist on allocating IPs to VMs though static addressing, DHCP is not allowed on production networks.

Reply:
Is that something that supported on other platforms?
I don't see in the API/YAML a way to supply IP per node,
Also, it contradicts the concept of the machine-API with machine-set, machine-set is like a cattle and not as a pet, it has a property 'replica' which is a number, you modify this number, and VMs are being created/destroyed.
I don't see how that can work with static IPs.
What am I missing here?